src/services/auth.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as firebase from "firebase-admin";
+import * as uno from "uno-serverless";
+import { memoize } from "uno-serverless";
+
+export type UserRecord = firebase.auth.UserRecord;
+
+export interface AuthService {
+  /**
+   * Initialize the service with the value of the HTTP Authorization header.
+   */
+  init(authorizationHeader: string): Promise<void>;
+
+  /**
+   * Return whether or not a user is authenticated.
+   */
+  isAuthenticated(): boolean;
+
+  /**
+   * Returns a promise that  which will resolve with the authenticated user.
+   *
+   * If no user is authenticated the promise will reject.
+   */
+   authenticatedUser(): Promise<UserRecord>;
+
+  /**
+   * Returns a promise which will resolve with the ID of the authenticated user.
+   *
+   * If no user is authenticated the promise will reject.
+   */
+  authenticatedUserId(): Promise<string>;
+}
+
+export class FirebaseAuthService implements AuthService {
+
+  private idToken?: firebase.auth.DecodedIdToken;
+
+  constructor(public readonly app: firebase.app.App) {}
```

Comment:
Where do we setup firebase app informations, we should use the same app  ?

Reply:
The app is initialized as _container service_ in `src/container.ts` see https://github.com/amavm/amavm-vdh-api/pull/9/files#diff-691ac3f0f06fddc7f3079b1a0f835425R113

In order to complete the work I will need our Firebase app config object. According to https://firebase.google.com/docs/web/setup#add_firebase_to_your_app it does not contain any secret so it could be added directly in the code.